### PR TITLE
Admins can change display name when flag set to false

### PR DESCRIPTION
**Pull request by sharidas**
When 'allow_user_to_change_display_name' => false, is
set in the config.php then admin user should be
able to set the display name.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

When 'allow_user_to_change_display_name' => false, is set in the config.php, then the admin user should be able to set the display name of users.

## Related Issue

https://github.com/owncloud/core/issues/27442

## Motivation and Context

When 'allow_user_to_change_display_name' => false, is set in the config.php, the admin user wasn't able to set the display name. This change will help to allow the admin to set the display name for users.

## How Has This Been Tested?

- [x] create 3 users admin, user1, user2
- [x] Add 'allow_user_to_change_display_name' => false, to the config.php
- [x] Login as admin user and try to rename the admin to "Foo". It worked.
- [x] Create a new group `test` and add `user1` and `user2` to the group `test`.
- [x] Make `user1` as the group admin for `test1`.
- [x] Now try to login as `user` then try to change the display name for `user1`, it worked.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sharidas** [comment]
Made modification so that admin and group admins can change the display name. It would be nice to know if we want to have only 
1) admins and group admins to change the display name of users
2) or **only** admins allowed to change display name

**Comment by DeepDiver1975** [comment]
Any chance to get this unit tested? THX

**Comment by PVince81** [comment]
@sharidas I think any kind of admin should be able to change the display name. Non-admins aka users should not.

**Comment by sharidas** [comment]
So if I understand correctly 
> admins and group admins to change the display name of users
Admins and group admins are allowed to change the display name if `allow_user_to_change_display_name' => false,` is added to config.php

**Comment by PVince81** [comment]
@sharidas correct

**Comment by PVince81** [comment]
@pmaier1 FYI ^

**Comment by sharidas** [comment]
Now an admin and group admin can change the users display name

**Comment by PVince81** [comment]
@sharidas please adjust unit tests. If there are none, please add some. Thanks

**Comment by sharidas** [comment]
> please adjust unit tests. If there are none, please add some. Thanks

Thats next task.

File: lib/private/User/User.php [hunk 1]
```diff
@@ -76,9 +83,12 @@ class User implements IUser {
 	 * @param IConfig|null $config
 	 * @param null $urlGenerator
 	 * @param EventDispatcher|null $eventDispatcher
+	 * @param Manager|null $groupManager
+	 * @param Session|null $userSession
 	 */
 	public function __construct(Account $account, AccountMapper $mapper, $emitter = null, IConfig $config = null,
-								$urlGenerator = null, EventDispatcher $eventDispatcher = null
+								$urlGenerator = null, EventDispatcher $eventDispatcher = null,
+								Manager $groupManager = null, Session $userSession = null
```
Comment (sharidas) [inline lib/private/User/User.php]: Injecting GroupManager and UserSession here. Didn't liked the previous version of the patch. So used this approach. Let me know if it looks ok.

Comment (sharidas) [reply lib/private/User/User.php]: Have also added unit test for the same.

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by sharidas** [comment]
Will create backport.

**Comment by sharidas** [comment]
Backport to stable10: https://github.com/owncloud/core/pull/29442

**Comment by DeepDiver1975** [comment]
```
16:07:28 [owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q] Running shell script
16:07:28 + export NOCOVERAGE=1
16:07:28 + NOCOVERAGE=1
16:07:28 + unset USEDOCKER
16:07:28 + phpenv local 5.6
16:07:28 + make test-php TEST_DATABASE=pgsql
16:07:28 PHPUNIT="/var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/phpunit/phpunit/phpunit" build/autotest.sh pgsql 
16:07:28 Using PHP executable /opt/phpenv/shims/php
16:07:28 Parsing all files in lib/public for the presence of @since or @deprecated on each method...
16:07:28 
16:07:30 Using database oc_autotest5
16:07:30 Setup environment for pgsql testing on local storage ...
16:07:34 Installing ....
16:07:52 PHP Fatal error:  Cannot use OC\Group\Manager as Manager because the name is already in use in /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/User/User.php on line 41
16:07:52 PHP Stack trace:
16:07:52 PHP   1. {main}() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/occ:0
16:07:52 PHP   2. require_once() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/occ:11
16:07:52 PHP   3. OC\Console\Application->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/console.php:106
16:07:52 PHP   4. Symfony\Component\Console\Application->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/Console/Application.php:162
16:07:52 PHP   5. Symfony\Component\Console\Application->doRun() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/symfony/console/Application.php:130
16:07:52 PHP   6. Symfony\Component\Console\Application->doRunCommand() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/symfony/console/Application.php:228
16:07:52 PHP   7. Symfony\Component\Console\Command\Command->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/symfony/console/Application.php:874
16:07:52 PHP   8. OC\Core\Command\Maintenance\Install->execute() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/symfony/console/Command/Command.php:264
16:07:52 PHP   9. OC\Setup->install() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/core/Command/Maintenance/Install.php:86
16:07:52 PHP  10. OC\User\Manager->createUser() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/Setup.php:347
16:07:52 PHP  11. OC\User\Manager->getUserObject() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/User/Manager.php:336
16:07:52 PHP  12. spl_autoload_call() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/User/Manager.php:180
16:07:52 PHP  13. Composer\Autoload\ClassLoader->loadClass() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/private/User/Manager.php:180
16:07:52 PHP  14. Composer\Autoload\includeFile() /var/lib/jenkins/workspace/owncloud-core_core_PR-29132-2PBBEYQ3IXYKPHPI4JLRKSIPJOYVTQEFY26WO6EJ72XKCBDAG24Q/lib/composer/composer/ClassLoader.php:322

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
https://github.com/owncloud/core/pull/29132#issuecomment-341735562

**Comment by sharidas** [comment]
All checks passed.

**Pull request merged**